tests/mail/tests.py

```diff
@@ -991,6 +991,22 @@ def test_empty_admins(self):
         mail_managers('hi', 'there')
         self.assertEqual(self.get_mailbox_content(), [])
 
+    def test_incorrect_settings(self):
+        settings = (
+            '123',
+            ('123'),
+            ['one@example.com', 'two@example.com'],
+        )
+        for value in settings:
+            with self.subTest(setting=value):
+                with self.settings(ADMINS=value, MANAGERS=value):
```

Comment:
You can join these two lines, e.g.
```python
tests = (
    'test@example.com',
    ('test@example.com',),
    ['test@example.com', 'other@example.com'],
    ('test@example.com', 'other@example.com'),
)
for setting, mail_func in (
    ('ADMINS', mail_admins),
    ('MANAGERS', mail_managers),
):
    msg = (
        'The %s setting must be a list or tuple containing two values.' % setting
    )
    for value in tests:
        with self.subTest(setting=setting, value=value), self.settings(**{setting: value}):
            with self.assertRaisesMessage(ValueError, msg):
                mail_func('subject', 'content')
```

Reply:
Shouldn't the error message be `.... a list of tuples`?

Reply:
Yes, sorry, typo `... a list of tuples containing two values.`